spec/requests/admin_payroll_run_downloads_spec.rb

```diff
@@ -12,5 +12,67 @@
       expect(response).to have_http_status(:ok)
       expect(response.body).to include("#{Date.today.strftime("%B")} payroll file")
     end
+
+    it "redirects if all payroll_runs have been downloaded" do
+      sign_in_to_admin_with_role(AdminSession::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE)
+
+      payroll_run = create(:payroll_run, downloaded_at: Time.zone.now)
+
+      expect(get(new_admin_payroll_run_download_path(payroll_run))).to redirect_to admin_payroll_run_download_path(payroll_run)
+    end
+  end
+
+  describe "downloads#show" do
+  it "redirects to the new action when payroll run has not been marked as downloaded" do
+    payroll_run = create(:payroll_run)
+
+    expect(get(admin_payroll_run_download_path(payroll_run))).to redirect_to new_admin_payroll_run_download_path(payroll_run)
+  end
+
+  it "shows a link to the download the payroll run file when payroll run has been marked as downloaded" do
+    payroll_run = create(:payroll_run, downloaded_at: Time.zone.now, downloaded_by: admin_session_id)
+
+    get admin_payroll_run_download_path(payroll_run)
+
+    expect(response.body).to include "Download the payroll run file for #{Date.today.strftime("%B")}."
+    expect(response.body).to include admin_payroll_run_download_path(payroll_run, format: :csv)
+  end
+
+  it "shows when and which user download the payroll run file when payroll run has been marked as downloaded" do
+    payroll_run = create(:payroll_run, downloaded_at: Time.zone.now, downloaded_by: "admin_user_id")
+
+    travel_to 31.seconds.from_now do
+      get admin_payroll_run_download_path(payroll_run)
+
+      expect(response.body).to include payroll_run.downloaded_by.to_s
+      expect(response.body).to include I18n.l(payroll_run.downloaded_at)
+    end
+  end
```

Comment:
Maybe also a spec that confirms that the CSV can be downloaded when the download is available?